coderd/coderd.go

```diff
@@ -627,6 +626,10 @@ func New(options *Options) *API {
 			// limit must be configurable by the admin.
 			apiRateLimiter,
 			httpmw.ReportCLITelemetry(api.Logger, options.Telemetry),
+			// CSRF only needs to apply to /api routes. It does not apply to GET requests
+			// anyway, which is most other routes. We want to exempt any external auth or
+			// application type routes.
+			httpmw.CSRF(options.SecureAuthCookie),
```

Comment:
There are a bunch of `ExemptRegexp()` calls inside this middleware that may need to be reviewed after this change. It might be nicer to have those passed explicitly as arguments to `httpmw.CSRF` instead of being hidden inside the middleware.

Reply:
The thing is, this will pretty quickly fail on someone when they add a new route if it breaks. The error is a CSRF error, which brings you to here. I prefer to have this all in 1 place, and this should never be configurable by a customer except maybe to disable it entirely.

I think the current exempts are overkill. I intentionally went a bit overkill because I was afraid of breaking anything. Some 


The reason there are exempts like this is when I turned this back on, I was trying to error on the side of caution. I think we should have a PR to remove an exempt 1 by 1 when we determine they can never cause an issue.

An example is the `/provisionerdaemons` ones. They are currently only GET requests, and I do not think the actual daemons are using cookies to authenticate. However I error-ed on the side of caution since these routes should never use CSRF anyway.

I can make follow up PRs to drop these with rational for each one